src/parser/tests/expression/NullCoalescenceExpression.spec.ts

```diff
@@ -374,5 +380,174 @@ describe('NullCoalescingExpression', () => {
                 end sub
             `);
         });
+
+        it('transpiles null coalescing in RHS of AssignmentStatement to if statement', () => {
+            testTranspile(`
+                sub main()
+                    a = user ?? {}
+                end sub
+            `, `
+                sub main()
+                    a = user
+                    if a = invalid then
+                        a = {}
+                    end if
+                end sub
+            `);
+        });
+
+        it('transpiles null coalescing in RHS of incrementor AssignmentStatement to if statement', () => {
+            testTranspile(`
+                sub main()
+                    a += user ?? 0
+                end sub
+            `, `
+                sub main()
+                    a += user
+                    if a = invalid then
+                        a += 0
+                    end if
+                end sub
+            `);
```

Comment:
This is incorrect. The logic should be:
`a += (user ?? 0)`

which would transpile to something more like:
```
if user
    a += user
else
    a += 0
end if
```

This also means we can't support this for dotted and indexed set statements without an intermediary variable (which I do not want to introduce at this point).